rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -141,14 +152,40 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
   }
 
   template<typename T>
-  typename std::enable_if<std::is_same<T, rcl_serialized_message_t>::value, void>::type
+  typename std::enable_if<
+    std::is_same<T, rcl_serialized_message_t>::value &&
+    !std::is_same<MessageT, rclcpp::experimental::SerializedContainer>::value,
+    void>::type
   execute_impl()
   {
-    throw std::runtime_error("Subscription intra-process can't handle serialized messages");
+    if (serializer_) {
+      rmw_message_info_t msg_info;
+      msg_info.from_intra_process = true;
+
+      ConstMessageSharedPtr msg = buffer_->consume_shared();
+      auto serialized_msg =
+        serializer_->serialize_message(reinterpret_cast<const void *>(msg.get()));
```

Comment:
do we need that `reinterpret_cast` here?

Reply:
to stasify the checks, that conversion to void* is intended